stats/thresholds_test.go

```diff
@@ -25,99 +25,274 @@ import (
 	"testing"
 	"time"
 
-	"github.com/dop251/goja"
 	"github.com/stretchr/testify/assert"
-
 	"go.k6.io/k6/lib/types"
 )
 
 func TestNewThreshold(t *testing.T) {
-	src := `1+1==2`
-	rt := goja.New()
+	t.Parallel()
+
+	src := `rate<0.01`
 	abortOnFail := false
 	gracePeriod := types.NullDurationFrom(2 * time.Second)
-	th, err := newThreshold(src, rt, abortOnFail, gracePeriod)
+	wantParsed := &thresholdExpression{tokenRate, tokenLess, 0.01}
+
+	gotThreshold, err := newThreshold(src, abortOnFail, gracePeriod)
+
 	assert.NoError(t, err)
+	assert.Equal(t, src, gotThreshold.Source)
+	assert.False(t, gotThreshold.LastFailed)
+	assert.Equal(t, abortOnFail, gotThreshold.AbortOnFail)
+	assert.Equal(t, gracePeriod, gotThreshold.AbortGracePeriod)
+	assert.Equal(t, wantParsed, gotThreshold.parsed)
+}
+
+func TestNewThreshold_InvalidThresholdConditionExpression(t *testing.T) {
+	t.Parallel()
+
+	src := "1+1==2"
+	abortOnFail := false
+	gracePeriod := types.NullDurationFrom(2 * time.Second)
+
+	gotThreshold, err := newThreshold(src, abortOnFail, gracePeriod)
+
+	assert.Error(t, err, "instantiating a threshold with an invalid expression should fail")
+	assert.Nil(t, gotThreshold, "instantiating a threshold with an invalid expression should return a nil Threshold")
+}
+
+func TestThreshold_runNoTaint(t *testing.T) {
+	t.Parallel()
+
+	type fields struct {
+		Source           string
+		LastFailed       bool
+		AbortOnFail      bool
+		AbortGracePeriod types.NullDuration
+		parsed           *thresholdExpression
+	}
+	type args struct {
+		sinks map[string]float64
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantOk  bool
+		wantErr bool
+	}{
+		{
+			name:    "valid expression using the > operator over passing threshold",
+			fields:  fields{"rate>0.01", false, false, types.NullDurationFrom(2 * time.Second), &thresholdExpression{tokenRate, tokenGreater, 0.01}},
+			args:    args{map[string]float64{"rate": 1}},
```

Comment:
Seems to me like you can just make `args` `map[string]float64` and skip some types and wrapping in the process

Reply:
Agreed 👍🏻